File: push/src/main/java/com/xiaomi/mipush/sdk/FCMPushHelper.java
package com.xiaomi.mipush.sdk;

import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;
import java.util.Map;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/FCMPushHelper.class */
public class FCMPushHelper {
    public static void clearToken(Context context) {
        AssemblePushHelper.clearToken(context, AssemblePush.ASSEMBLE_PUSH_FCM);
    }

    public static void convertMessage(Intent intent) {
        AssemblePushHelper.convertMessage(intent);
    }

    public static boolean isFCMSwitchOpen(Context context) {
        return AssemblePushHelper.isOpenAssemblePushOnlineSwitch(context, AssemblePush.ASSEMBLE_PUSH_FCM) && MiPushClient.getOpenFCMPush(context);
    }

    public static void notifyFCMNotificationCome(Context context, Map<String, String> map) {
        PushMessageReceiver miPushReceiver;
        String str = map.get("pushMsg");
        if (TextUtils.isEmpty(str) || (miPushReceiver = AssemblePushHelper.getMiPushReceiver(context)) == null) {
            return;
        }
        miPushReceiver.onNotificationMessageArrived(context, AssemblePushHelper.parseMiPushMessage(str));
    }

    public static void notifyFCMPassThoughMessageCome(Context context, Map<String, String> map) {
        PushMessageReceiver miPushReceiver;
        String str = map.get("pushMsg");
        if (TextUtils.isEmpty(str) || (miPushReceiver = AssemblePushHelper.getMiPushReceiver(context)) == null) {
            return;
        }
        miPushReceiver.onReceivePassThroughMessage(context, AssemblePushHelper.parseMiPushMessage(str));
    }

    public static void reportFCMMessageDelete() {
        MiTinyDataClient.upload(AssemblePushHelper.getSPErrorKey(AssemblePush.ASSEMBLE_PUSH_FCM), "fcm", 1L, "some fcm messages was deleted ");
    }

    public static void uploadToken(Context context, String str) {
        AssemblePushHelper.uploadToken(context, AssemblePush.ASSEMBLE_PUSH_FCM, str);
    }
}

File: legacy/src/main/java/com/xiaomi/mipush/sdk/FCMPushHelper.kt
package com.xiaomi.mipush.sdk

import android.content.Context
import android.content.Intent
import android.text.TextUtils

/*
 * Local compatibility reference:
 * this helper has no same-path source in com.xiaomi.xmsf 7.4.67-C stock or the
 * 2026-04-13 current override tree, and is retained to preserve the legacy
 * `com.xiaomi.mipush.sdk` FCM surface used by the app-facing compatibility layer.
 */
object FCMPushHelper {
    @JvmStatic
    fun clearToken(context: Context) {
        AssemblePushHelper.clearToken(context, AssemblePush.ASSEMBLE_PUSH_FCM)
    }

    @JvmStatic
    fun convertMessage(intent: Intent) {
        AssemblePushHelper.convertMessage(intent)
    }

    @JvmStatic
    fun isFCMSwitchOpen(context: Context): Boolean {
        return AssemblePushHelper.isOpenAssemblePushOnlineSwitch(context, AssemblePush.ASSEMBLE_PUSH_FCM) &&
            MiPushClient.getOpenFCMPush(context)
    }

    @JvmStatic
    fun notifyFCMNotificationCome(context: Context, map: Map<String, String>) {
        val pushMsg = map["pushMsg"]
        val receiver = AssemblePushHelper.getMiPushReceiver(context)
        if (TextUtils.isEmpty(pushMsg) || receiver == null) {
            return
        }
        receiver.onNotificationMessageArrived(context, AssemblePushHelper.parseMiPushMessage(pushMsg ?: ""))
    }

    @JvmStatic
    fun notifyFCMPassThoughMessageCome(context: Context, map: Map<String, String>) {
        val pushMsg = map["pushMsg"]
        val receiver = AssemblePushHelper.getMiPushReceiver(context)
        if (TextUtils.isEmpty(pushMsg) || receiver == null) {
            return
        }
        receiver.onReceivePassThroughMessage(context, AssemblePushHelper.parseMiPushMessage(pushMsg ?: ""))
    }

    @JvmStatic
    fun reportFCMMessageDelete() {
        MiTinyDataClient.upload(
            AssemblePushHelper.getSPErrorKey(AssemblePush.ASSEMBLE_PUSH_FCM) ?: "",
            "fcm",
            1L,
            "some fcm messages was deleted "
        )
    }

    @JvmStatic
    fun uploadToken(context: Context, token: String) {
        AssemblePushHelper.uploadToken(context, AssemblePush.ASSEMBLE_PUSH_FCM, token)
    }
}
